pkg/cnbutils/privacy/privacy.go

```diff
@@ -0,0 +1,47 @@
+package privacy
+
+import (
+	"strings"
+
+	containerName "github.com/google/go-containerregistry/pkg/name"
+)
+
+func FilterBuildpacks(buildpacks []string) []string {
+	result := make([]string, 0, len(buildpacks))
+	for _, buildpack := range buildpacks {
+		ref, err := containerName.ParseReference(strings.ToLower(buildpack))
+		if err != nil {
+			result = append(result, "<error>")
+			continue
+		}
+
+		registry := ref.Context().Registry.Name()
+		repository := ref.Context().RepositoryStr()
+
+		if registry == "gcr.io" && strings.HasPrefix(repository, "paketo-buildpacks/") {
+			result = append(result, buildpack)
+		} else if registry == "index.docker.io" && strings.HasPrefix(repository, "paketobuildpacks/") {
+			result = append(result, buildpack)
+		} else if registry == "gcr.io" && strings.HasPrefix(repository, "buildpacks/") {
+			result = append(result, buildpack)
+		} else if registry == "public.ecr.aws" && strings.HasPrefix(repository, "heroku-buildpacks/") {
+			result = append(result, buildpack)
```

Comment:
Should we move the pairs of registry and repo prefix into an array that we just iterate over instead of if-else-if-else-if-...?

Reply:
Also, istn't the registry just part of the prefix, if we e.g. would use `ref.Name()` instead of first splitting this up into `ref.Context().Registry.Name()` and `ref.Context().RepositoryStr()`. I.e. we could have an array of prefix **strings** as allow-list...

Reply:
> Should we move the pairs of registry and repo prefix into an array that we just iterate over instead of if-else-if-else-if-...?

See https://github.com/SAP/jenkins-library/pull/3368/commits/36df0e7327d57d10f1533318dde118ffd6e29a71



> Also, istn't the registry just part of the prefix, if we e.g. would use `ref.Name()` instead of first splitting this up into `ref.Context().Registry.Name()` and `ref.Context().RepositoryStr()`. I.e. we could have an array of prefix **strings** as allow-list...

We relay on the normalization it does for Docker Hub (turning every alias into `index.docker.io`). For some reason we only get this when using `ref.Context().Registry.Name()`